### PR TITLE
update python sdk (December 2021)

### PR DESCRIPTION
**Pull request by leewang0**
## Description of the change

> Adds the latest updates (some new endpoints, new models) from Merge's HRIS API as defined via OpenAPI here: https://api.merge.dev/api/hris/v1/schema

> Importantly also includes a few tests for various HRIS models.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests, or the description above explains how testing was performed

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or gif attached. Download "Kap" from the Mac App store for easy gif screen capture.

### CODE REVIEW

**Review by BrendanGoggin — APPROVED** [review]

File: MergeHRISClient/model_utils.py [hunk 1]
```diff
@@ -113,6 +113,12 @@ def set_attribute(self, name, value):
 
         if name in self.openapi_types:
             required_types_mixed = self.openapi_types[name]
+        elif self.additional_properties_type is None:
+            raise ApiAttributeError(
+                "{0} has no attribute '{1}'".format(
+                    type(self).__name__, name),
+                path_to_item
+            )
```
Comment (dan-merge) [inline MergeHRISClient/model_utils.py:121]: i dont think we want this in here, pretty sure it causes issues

File: MergeHRISClient/model_utils.py [hunk 2]
```diff
@@ -1488,10 +1494,19 @@ def model_to_dict(model_instance, serialize=True):
...
+               else:
+                   res = []
+                   for v in value:
+                       if isinstance(v, PRIMITIVE_TYPES) or v is None:
+                           res.append(v)
+                       elif isinstance(v, ModelSimple):
+                           res.append(v.value)
+                       else:
+                           res.append(model_to_dict(v, serialize=serialize))
+                   result[attr] = res
```
Comment (dan-merge) [inline MergeHRISClient/model_utils.py:1509]: ^ is this from a new generator version?

Comment (leewang0) [reply MergeHRISClient/model_utils.py:1509]: Yeah i was surprised that this file had changed, i definitely use the same generator version https://github.com/merge-api/merge-backend/blob/develop/docker-compose.yml#L112 of `5.1.1`

Comment (dan-merge) [reply MergeHRISClient/model_utils.py:1509]: tbh we should probably git ignore this file unless we have explicit changes we want to make, it causes a lot of problems

Comment (leewang0) [reply MergeHRISClient/model_utils.py]: All set, although I'm going to hold off on the gitignore for now

**Pull request merged**